### DISCUSSION TITLE
All games stats

### DISCUSSION THREAD

**Discussion by sienmich** [bafolts/terraforming-mars#2119] [Ideas]
I would love to have collected game statistics and a nice way to view it.

For example for every card (including preludes and corps):
- it's play ratio (how many times this cards was played divided by all games). 
- it's win ratio (how many games were won by player that played this card divided by all the games that played this card).
- it's average points (mean of final VP from players that played this card; maybe table for each total generations number) 

It could be easily done on current servers, but there are downsides:
- There are a lot of servers, so each would have different stats
- The main server is restarted daily

So maybe there would be a way to send those finished games jsons to some other server?
Or could we somehow store tiny version of game results on main server? How much space consuming it would be, how much space in database we have?

How many games there are on main daily?

Would you like those statistics? Any others?

**Comment by kevinb9n** [comment]
Let me just say I love this kind of thing and it would be cool as hell to get there. I think #1110 is maybe the relevant issue for the json you're talking about wanting. It would be cool to have an analysis server any game server could feed those to.

Things that have crossed my mind:

* How early or late in the game does this card get played? (Suppose we represent each mandatory GP as a number from 0-1 and average those, call that value "time". Or just do `(gen-1)/(totalGens-1)`.) (Example, do people "always" hang onto TerraGany til the very end?)
* How much does the card "pay out" vs. the time it was played? (Total resources it generated or saved) (Example, does Marineris pay out 1.9 ocean tiles, or 1.6, or what? How much MC does Research Outpost save on average)
* For every "OR" instruction, how many times was each component used? (maybe this is mostly redundant if the payout is being calculated)

**Comment by ltdstudio** [comment]
My playmates and I would love to have a ranking.
